rskj-core/src/test/java/co/rsk/peg/BridgeSupportTestUtil.java

```diff
@@ -22,6 +24,55 @@ public static Repository createRepository() {
         return new MutableRepository(new MutableTrieCache(new MutableTrieImpl(null, new Trie())));
     }
 
+    public static PartialMerkleTree createValidPmtForTransactions(List<Sha256Hash> hashesToAdd, NetworkParameters networkParameters) {
+        byte[] relevantNodesBits = new byte[(int)Math.ceil(hashesToAdd.size() / 8.0)];
+        for (int i = 0; i < hashesToAdd.size(); i++) {
+            Utils.setBitLE(relevantNodesBits, i);
+        }
+
+        return PartialMerkleTree.buildFromLeaves(networkParameters, relevantNodesBits, hashesToAdd);
+    }
+
+    public static void recreateChainFromPmt(
```

Comment:
Should this return something? I guess the idea now is that it modifies the received `btcBlockStoreWithCache` object right? 

But is there any reason not to simply create a new one and return it?

Reply:
because to create it you need a lot of parameters (repository, bridgeConstants, bridgeStorageProvider, activations, etc. See [here](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/RepositoryBtcBlockStoreWithCache.java#L65)).
This way, in the tests we set the btcBlockStoreFactory and a new btcBlockStoreWithCache instance from it, using the parameters from the test